packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -0,0 +1,318 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as core from '@aws-cdk/core';
+import { ClusterParameterGroup, IParameterGroup, ParameterGroup } from './parameter-group';
+import { ParameterGroupFamilyMapping } from './private/parameter-group-family-mapping';
+import { compare } from './private/version';
+
+/**
+ * The extra options passed to the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineBindOptions {
+  /**
+   * The role used for S3 importing.
+   *
+   * @default - none
+   */
+  readonly s3ImportRole?: iam.IRole;
+
+  /**
+   * The role used for S3 exporting.
+   *
+   *  @default - none
+   */
+  readonly s3ExportRole?: iam.IRole;
+
+  /**
+   * The customer-provided ParameterGroup.
+   *
+   * @default - none
+   */
+  readonly parameterGroup?: IParameterGroup;
+}
+
+/**
+ * The type returned from the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineConfig {
+  /**
+   * The ParameterGroup to use for the cluster.
+   *
+   * @default - no ParameterGroup will be used
+   */
+  readonly parameterGroup?: IParameterGroup;
+
+  /**
+   * The port to use for this cluster,
+   * unless the customer specified the port directly.
+   *
+   * @default - use the default port for clusters (3306)
+   */
+  readonly port?: number;
+}
+
+/**
+ * The interface representing a database cluster (as opposed to instance) engine.
+ */
+export interface IClusterEngine {
+  /** The type of the engine, for example "aurora-mysql". */
+  readonly engineType: string;
```

Comment:
Should this be better typed than just string. There's a limited set of allowed values here.

Reply:
I kept this out of this PR in order to make it smaller, as explained here : https://github.com/aws/aws-cdk/pull/8686#issuecomment-647773881

Reply:
👍